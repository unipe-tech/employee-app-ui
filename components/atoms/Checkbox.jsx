import React from "react";
import { StyleSheet,Text, View } from "react-native";
import CheckBox from "@react-native-community/checkbox";

import { COLORS, FONTS } from "../../constants/Theme";
import { checkBox } from "../../styles";

const Checkbox = ({ value, setValue, text }) => (
    <View style={styles.container}>
      <CheckBox
        value={value}
        onValueChange={setValue}
        tintColors={{ true: COLORS.primary }}
      />
      <Text style={styles.title}>{text}</Text>
    </View>
  );

const styles = StyleSheet.create({
  container: {
    flexDirection: "row",
    alignItems: "center",
    marginVertical: 10,
  },
  title: {
    marginLeft: 5,
    ...FONTS.h4,
    color: COLORS.gray,
    flex: 1,
  },
});

export default Checkbox;
