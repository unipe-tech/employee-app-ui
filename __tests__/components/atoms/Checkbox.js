import { render, fireEvent, waitFor } from "@testing-library/react-native";
import { describe, expect, test } from "@jest/globals";
import Checkbox from "../../../components/atoms/Checkbox";

describe("Checkbox", () => {
  test("renders UI correctly", () => {
    const tree = render(<Checkbox />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
