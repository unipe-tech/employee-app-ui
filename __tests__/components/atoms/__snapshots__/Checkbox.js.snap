// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox renders UI correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "marginVertical": 19.736842105263158,
    }
  }
>
  <View
    accessibilityRole="checkbox"
    accessibilityState={
      Object {
        "checked": false,
        "disabled": false,
      }
    }
    accessibilityValue={
      Object {
        "text": "off",
      }
    }
    accessible={true}
    pointerEvents="auto"
  >
    <RNCCheckbox
      forwardedRef={null}
      onValueChange={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "height": 32,
            "width": 32,
          },
          undefined,
        ]
      }
      tintColors={
        Object {
          "true": "#2CB77C",
        }
      }
    />
  </View>
  <Text
    style={
      Object {
        "color": "#808080",
        "flex": 1,
        "fontFamily": "Montserrat-Bold",
        "fontSize": 14,
        "lineHeight": 18,
        "marginLeft": 9.868421052631579,
      }
    }
  />
</View>
`;
